File: WPF/MVVM/Converters/SubstringConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Namespace Common.MVVM.Converters

    Public Class SubstringConverter
        Implements IValueConverter

        Public Function Convert(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim P = DirectCast(Parameter, String).Split(","c).Select(Function(S) Integer.Parse(S.Trim())).ToArray()
            Verify.True(P.Length = 2)
            Return CType(Value, String).Substring(P(0), P(1))
        End Function

        Public Function ConvertBack(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotSupportedException()
        End Function

    End Class

End Namespace

File: WPF/MVVM/Converters/SubstringConverter.cs
using System.Data;
using System.Linq;
using System;
using System.Globalization;
using System.Windows.Data;

namespace Ks
{
    namespace Common.MVVM.Converters
    {
        public class SubstringConverter : IValueConverter
        {
            public object Convert(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                var P = ((string)Parameter).Split(',').Select(S => int.Parse(S.Trim())).ToArray();
                Verify.True(P.Length == 2);
                return System.Convert.ToString(Value).Substring(P[0], P[1]);
            }

            public object ConvertBack(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                throw new NotSupportedException();
            }
        }
    }
}
